webapi/core.py

```diff
@@ -1,20 +1,95 @@
 import logging
 
+from deluge import component
 from deluge.plugins.pluginbase import CorePluginBase
+from deluge.ui.web.json_api import JSON
+from twisted.web import http, server
+import deluge.configmanager
+from deluge.core.rpcserver import export
 
 
 LOGGER = logging.getLogger(__name__)
 
+DEFAULT_PREFS = {
+    "enable_cors"         : False,
+    "allowed_origin"      : []
+}
 
 class Core(CorePluginBase):
 
     def enable(self):
         """"""
         LOGGER.info('Enabling WebAPI plugin CORE ...')
+        self.JSON_instance = component.get('JSON')
+        self.patched = False
+        self.config = deluge.configmanager.ConfigManager("webapi.conf", DEFAULT_PREFS)
+        if (self.config["enable_cors"] == True):
+            self.patch_web_ui()
 
     def disable(self):
         """"""
         LOGGER.info('Disabling WebAPI plugin CORE ...')
+        self.config.save()
 
     def update(self):
         """"""
+
+    @export
+    def set_config(self, config):
+        "sets the config dictionary"
+        for key in config.keys():
+            self.config[key] = config[key]
```

Comment:
That seems to be the same as `dict.update(dict)`

Reply:
This is not possible, as self.config is of type deluge.configmanager.ConfigManager and not directly a dict